firebase-vscode/src/data-connect/service.ts

```
@@ -78,47 +81,35 @@ export class DataConnectService {
     return response.text();
   }
   private async handleProdResponse(
-    clientResponse: ClientResponse<
+    response: ClientResponse<
       ExecuteGraphqlResponse | ExecuteGraphqlResponseError
     >,
   ): Promise<ExecutionResult> {
-    if (!(clientResponse.status >= 200 && clientResponse.status < 300)) {
+    if (!(response.status >= 200 && response.status < 300)) {
       const errorResponse =
-        clientResponse as ClientResponse<ExecuteGraphqlResponseError>;
+        response as ClientResponse<ExecuteGraphqlResponseError>;
       throw new DataConnectError(
-        `Request failed with status ${clientResponse.status}`,
-        errorResponse.body.error.message,
+        `Prod Request failed with status ${response.status}\nMessage ${errorResponse?.body?.error?.message}`,
       );
     }
-    const successResponse =
-      clientResponse as ClientResponse<ExecuteGraphqlResponse>;
+    const successResponse = response as ClientResponse<ExecuteGraphqlResponse>;
     return successResponse.body;
   }
 
-  private async handleValidResponse(
-    response: Response,
+  private async handleEmulatorResponse(
+    response: ClientResponse<
+      ExecuteGraphqlResponse | ExecuteGraphqlResponseError
+    >,
   ): Promise<ExecutionResult> {
-    const json = await this.decodeResponse(response, "application/json");
-    assertExecutionResult(json);
-
-    return json;
-  }
-
-  private async handleInvalidResponse(response: Response): Promise<never> {
-    const cause = await this.decodeResponse(response);
-
-    throw new DataConnectError(
-      `Request failed with status ${response.status}`,
```

It swallowed the error message here.